### ISSUE TITLE
Calculate topic similarity from multilingual topic models

### ISSUE THREAD

**Issue by njuljw76** [MaartenGr/BERTopic#851]
Hi, thank you for BERTopic. It is really wonderful🥳👍

I have a question:

I have two datasets in different languages, I want to train separate BERTopic models and compare their overlap between topics in model A and model B.

I read the ‘tips_and_tricks’ file in your docs, and the part ‘Finding similar topics between models’ of this is perfect for my needs. You give two methods and give a sample code for the first method, we can use ‘topic_embeddings’ which are c-TF-IDF weighted embeddings based on the top n words to calculate similarity. 

But the topic_embeddings is obtained after dimensionality reduction, umap is a stochastic algorithm and thus generates different results every time, the results of dimensionality reduction for two datasets will be in a different space.
Does it make sense to use cosine similarity to calculate the embeddings of two datasets when the topic_embeddings obtained after dimensionality reduction do not have the same units and space?

Kind regards,
Liang~

**Comment by MaartenGr** [comment]
> Does it make sense to use cosine similarity to calculate the embeddings of two datasets when the topic_embeddings obtained after dimensionality reduction do not have the same units and space?

As you mention, the `.topic_embeddings_` are indeed c-TF-IDF weighted embeddings. These embeddings are generated by taking the top n words and passing them through the embedding model. In other words, the `.topic_embeddings_` are not obtained as a result of the dimensionality reduction. Although it is a sequential process, there are essential two parts to BERTopic. First is the clustering step and then the topic representation step (e.g., through `.topic_embeddings_`). These two steps are somewhat independent of one another, so the dimensionality reduction step does not influence the dimensional space of the `.topic_embeddings_`. 

In other words, since the `.topic_embeddings_` from the two datasets are generated independently from the dimensionality reduction step, they will have the same dimensional space as long as you use the same embedding model.

**Comment by njuljw76** [comment]
Thanks for replying so quickly and patience! Your explanation is very clear! It looks like there was some error in my previous understanding, and now I have a clearer idea of it! Thank you!
